experimental/JuLie/quantum_analogs.jl

```diff
@@ -0,0 +1,295 @@
+q^3 + 2*q^2 + 2*q + 1
+
+julia> quantum_factorial(3,2)
+21
+
+julia> K,i = cyclotomic_field(4, "i");
+
+julia> quantum_factorial(3, i)
+i - 1
+```
+"""
+function quantum_factorial(n::IntegerUnion, q::RingElem)
+
+	n >= 0 || throw(ArgumentError("n ≥ 0 required"))
+
+	R = parent(q)
+	if isone(q)
+		return R(factorial(n))
+	else
+		z = one(R)
+		for i = 1:n
+			z *= quantum_integer(i,q)
+		end
+		return z
+	end
+end
+
+function quantum_factorial(n::IntegerUnion, q::Integer)
+	return quantum_factorial(n,ZZ(q))
+end
+
+function quantum_factorial(n::IntegerUnion)
+	R,q = LaurentPolynomialRing(ZZ, "q")
+	return quantum_factorial(n,q)
+end
+
+
+################################################################################
+# Quantum binomials
+################################################################################
+
+@doc Markdown.doc"""
+    quantum_binomial(n::IntegerUnion, q::RingElem)
+    quantum_binomial(n::IntegerUnion, k::IntegerUnion, q::Integer)
+    quantum_binomial(n::IntegerUnion, k::IntegerUnion)
+
+Let ``k`` be a non-negative integer and let ``n ∈ ℤ``. The **quantum binomial**
+``\begin{bmatrix} n \\ k \end{bmatrix}_𝐪 \in ℚ(𝐪)`` is defined as
+```math
+\begin{bmatrix} n \\ k \end{bmatrix}_𝐪 ≔ \frac{[n]_𝐪!}{[k]_𝐪! [n-k]_𝐪!} = \frac{[n]_𝐪 [n-1]_𝐪⋅ … ⋅ [n-k+1]_𝐪}{[k]_𝐪!}
+```
+Note that the first expression is only defined for ``n ≥ k`` since the quantum factorials
+are only defined for non-negative integers—but the second  expression is well-defined for
+all ``n ∈ ℤ`` and is used for the definition. In [Con00](@cite) it is shown that
+```math
+\begin{bmatrix} n \\ k \end{bmatrix}_𝐪 = \sum_{i=0}^{n-k} q^i \begin{bmatrix} i+k-1 \\ k-1 \end{bmatrix}_𝐪 \quad \text{if } n ≥ k > 0 \;.
+```
+Since
+```math
+\begin{bmatrix} n \\ 0 \end{bmatrix}_𝐪 = 1 \quad \text{for all } n ∈ ℤ
+```
+and
+```math
+\begin{bmatrix} n \\ k \end{bmatrix}_𝐪 = 0 \quad \text{if } 0 ≤ n < k \;,
+```
+it follows inductively that
+```math
+\begin{bmatrix} n \\ k \end{bmatrix}_𝐪 ∈ ℤ[𝐪] \quad \text{if } n ≥ 0 \;.
+```
+For all ``n ∈ ℤ`` we have the relation
+```math
+\begin{bmatrix} n \\ k \end{bmatrix}_𝐪 = (-1)^k 𝐪^{-k(k-1)/2+kn} \begin{bmatrix} k-n-1 \\ k \end{bmatrix}_𝐪 \;,
+```
+which shows that
+```math
+\begin{bmatrix} n \\ k \end{bmatrix}_𝐪 ∈ ℤ[𝐪^{-1}] \quad \text{if } n < 0 \;.
+```
+In particular,
+```math
+\begin{bmatrix} n \\ k \end{bmatrix}_𝐪 ∈ ℤ[𝐪,𝐪^{-1}] \quad \text{for all } n ∈ ℤ \;.
+```
+Now, for an element ``q`` of a ring ``R`` we define ``\begin{bmatrix} n \\ k
+\end{bmatrix}_q`` as the specialization of ``\begin{bmatrix} n \\ k
+\end{bmatrix}_{\mathbf{q}}`` in ``q``, where ``q`` is assumed to be invertible in ``R`` if
+``n < 0``.
+
+Note that for ``q=1`` we obtain
+```math
+\begin{bmatrix} n \\ k \end{bmatrix}_1 = {n \choose k} \;,
+```
+hence the quantum binomial coefficient is a "deformation" of the usual binomial coefficient.
+
+# Functions
+
+The functions `quantum_binomial` work analogously to [`quantum_integer`](@ref).
+
```

Comment:
This section doesn't really add anything, esp. if the text before it were shortened to just describe what the function does?
```suggestion
```